src/util/Range.cpp

```diff
@@ -1,33 +1,47 @@
 #include "util/Range.h"
 
 #include <algorithm>  // for max, min
+#include <limits>     // for max, lowest
 
-Range::Range(double x, double y) {
-    this->x1 = x;
-    this->x2 = x;
+#include "util/Rectangle.h"
 
-    this->y1 = y;
-    this->y2 = y;
-}
+Range::Range():
+        minX(std::numeric_limits<double>::max()),
+        minY(std::numeric_limits<double>::max()),
+        maxX(std::numeric_limits<double>::lowest()),
+        maxY(std::numeric_limits<double>::lowest()) {}
+
+Range::Range(double x, double y): minX(x), minY(y), maxX(x), maxY(y) {}
```

Comment:
Move to header, since this does not rely on another header file (faster compile time).
```suggestion
```